controller/controller.go

```diff
@@ -708,69 +714,103 @@ func (s *Service) heartbeatController(ctx context.Context, addr *url.URL) {
 }
 
 func (s *Service) watchModuleChanges(ctx context.Context, sendChange func(response *ftlv1.PullSchemaResponse) error) error {
-	moduleSchemas := map[string]schema.Module{}
+	logger := log.FromContext(ctx)
+	type moduleStateEntry struct {
+		hash        sha256.SHA256
```

Comment:
@wesbillman Instead of storing the full schema struct I store a hash and just compare that, as it dawned on me that we don't actually care about anything in the schema.